tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -325,6 +326,38 @@ private void FeaturesProductConfigChanged(object sender, ProductConfigChangedEve
             e.Acknowledge(features.Name);
         }
 
+        private void AsmDataProductConfigRemoved(object sender, ProductConfigChangedEventArgs e)
+        {
+            if (!_enabled)
+            {
+                return;
+            }
+
+            var asmDataConfigs = e.GetDeserializedConfigurations<RcmModels.AsmData.Payload>();
+            foreach (var asmDataConfig in asmDataConfigs)
+            {
+                if (_remoteConfigurationStatus.RulesDataByFile.ContainsKey(asmDataConfig.Name))
```

Comment:
Nit: Dictionary.Remove(key) already checks if the key exists and can return true if the key is present or false otherwise. If it is not present, no exception will be thrown so, I think that the containsKey check could be skipped to avoid double checking that.